visionmetrics/detection/mean_ap.py

```diff
@@ -34,14 +34,19 @@ def __init__(self, box_format='xyxy', **kwargs):
         if box_format != 'xyxy':
             raise ValueError(f'Expected box format to be "xyxy", got {box_format}')
         super().__init__(box_format=box_format, **kwargs)
+        self._label_ids = set()
```

Comment:
There is reset and clone you might want to make sure they are working properly after adding this, with unit tests.

Reply:
https://github.com/Lightning-AI/torchmetrics/blob/3b3b997487fea2e05a65d81abc23a7ebef3371f1/src/torchmetrics/metric.py#L658

Reply:
added a unit test to verify label_ids set across batches.

Reply:
**Clone test**
a = metric()
a.update(...., ....)
b = a.clone()
assert b.calculate() == a.calculate()

**Reset**
a = metric()
a.update(data)
metric_val = a.calculate()
a.reset()
a.update(data)
assert a.calculate() == metric_val

Reply:
Updated. 

Now using existing *-labels attributes of the Metric class to track all 'seen' labels. Added unit tests.